### PR TITLE
PPP-2913 Add Startup probe type

### PR DESCRIPTION
**Pull request by mgruesen**
Add support for Startup probe type

https://jira.namely.land/browse/PPP-2913

### CODE REVIEW

File: pipeline/builder/types/types.go [hunk 1]
```diff
@@ -332,6 +332,7 @@ type Container struct {
 
 	LivenessProbe  *Probe `json:"livenessProbe"`
 	ReadinessProbe *Probe `json:"readinessProbe"`
+	StartupProbe   *Probe `json:"startupProbe"`
```
Comment (petermcneely) [inline pipeline/builder/types/types.go:335]: neat! I'm glad to see this being added. We're going to need to do a bit more work to support this however. Basically, you'll need to handle mapping the startup probe's definition from the yml file over to the json payload that gets sent to spinnaker.

Comment (mgruesen) [reply pipeline/builder/types/types.go:335]: Great, thanks for the guidance. I'll be making a Jira ticket to work that part then

Comment (mgruesen) [reply pipeline/builder/types/types.go:335]: @petermcneely Where does the mapping happen? The startup probe shares the same structure as the others, fyi

**Review by nycdotnet — APPROVED** [review]
neat!!

**Pull request merged**